src/psyclone/tests/psyir/symbols/symboltable_test.py

```diff
@@ -406,7 +589,80 @@ def test_lookup_visibility_filter():
             "searching for symbol 'var2'"in str(err))
 
 
-def test_lookup_with_tag():
+def test_lookup_3():
+    '''Check that lookup() in the SymbolTable class raises the expected
+    exception if the name or check_ancestors argument have the wrong type.
+
+    '''
+    sym_table = SymbolTable()
+    symbol = DataSymbol("var1", REAL_TYPE)
+    with pytest.raises(TypeError) as info:
+        _ = sym_table.lookup(symbol)
+    assert ("Expected the name argument to the lookup() method to be a str "
+            "but found 'DataSymbol'." in str(info.value))
+
+    sym_table = SymbolTable()
+    symbol = DataSymbol("var1", REAL_TYPE)
+    with pytest.raises(TypeError) as info:
+        _ = sym_table.lookup("var1", check_ancestors="hello")
+    assert ("Expected the check_ancestors optional argument to the lookup() "
+            "method to be a boolean but found 'str'." in str(info.value))
+
+
+def test_lookup_4():
+    '''Check that lookup() in the SymbolTable class behaves as
+    expected with the check_ancestors flag being a) explicitly set to
+    False, b) explicitly set to True and c) using the default value
+    (True).
+
+    '''
+    schedule_symbol_table = SymbolTable()
```

Comment:
Replace with utility routine?

Reply:
Yes, done as part of response to earlier comment (and for all other tests with the same pattern).